src/common/database.c

```diff
@@ -4391,6 +4393,55 @@ gchar *dt_database_get_most_recent_snap(const char* db_filename)
   return ret;
 }
 
+// nested transactions support
+
+static int trxid = 0;
```

Comment:
Hum... If we access the database with different threads that all our transaction are currently broken as we could have interleaved BEGIN/COMMIT from different thread. No?

Reply:
> If we access the database with different threads that all our transaction are currently broken as we could have interleaved BEGIN/COMMIT from different thread. No?

I think yes, at least with the same handle. But, until now, there may (almost) not be cross over.

> I have the trxid going below 0 when importing pictures

That's about nested BEGIN/COMMITs. That would mean there are more exits than entries somewhere ...

Reply:
But a single COMMIT will commit all the current transactions. And since we have interleaved transactions due to threading I don't see how to fix this mess. There was not error reported before because as I said a COMMIT do commit whatever is already in a transaction.

Reply:
I will at least try to understand where we have unmatched BEGIN/COMMIT. This will have at least helped!